opensearch-operator/pkg/reconcilers/cluster.go

```diff
@@ -103,10 +110,63 @@ func (r *ClusterReconciler) Reconcile() (ctrl.Result, error) {
 		})
 		result.CombineErr(err)
 	}
-
+	if r.instance.Spec.General.Snapshot != nil && len(r.instance.Spec.General.Snapshot) > 0 {
+		// Calculate checksum and check for changes
+		result.Combine(r.ReconcileSnapshotConfig(username))
+	}
 	return result.Result, result.Err
 }
 
+func (r *ClusterReconciler) ReconcileSnapshotConfig(username string) (*ctrl.Result, error) {
+	annotations := map[string]string{"cluster-name": r.instance.GetName()}
+	var checksumerr error
+	var checksumval string
+	snapshotSetting := batchv1.Job{}
+	snapshotdata, _ := json.Marshal(&r.instance.Spec.General.Snapshot)
+	checksumval, checksumerr = util.GetSha1Sum(snapshotdata)
+	if checksumerr != nil {
+		return &ctrl.Result{}, checksumerr
+	}
+	clusterName := r.instance.Name
+	jobName := clusterName + "-snapshotconfig-update"
+	job := batchv1.Job{}
+	if err := r.Get(r.ctx, client.ObjectKey{Name: jobName, Namespace: r.instance.Namespace}, &job); err == nil {
+		value, exists := job.ObjectMeta.Annotations[snapshotChecksumAnnotation]
+		if exists && value == checksumval {
+			// Nothing to do, current snapshotconfig already applied
+			return &ctrl.Result{}, nil
+		}
+		// Delete old job
+		r.logger.Info("Deleting old snapshotconfig job")
+		opts := client.DeleteOptions{}
+		// Add this so pods of the job are deleted as well, otherwise they would remain as orphaned pods
+		client.PropagationPolicy(metav1.DeletePropagationForeground).ApplyToDelete(&opts)
+		err = r.Delete(r.ctx, &job, &opts)
+		if err != nil {
+			return &ctrl.Result{}, err
+		}
+		// Make sure job is completely deleted (when r.Delete returns deletion sometimes is not yet complete)
+		_, err = r.ReconcileResource(&job, reconciler.StateAbsent)
+		if err != nil {
+			return &ctrl.Result{}, err
+		}
+	}
+	r.logger.Info("Starting snapshotconfig update job")
```

Comment:
So just for understating, you have a snapshot _update_configuration job that declares new jobs from the CRD,  if one snapshot declaration has removed from the CRD, how do you detect that?

Reply:
@idanl21 the removing of a snapshot repo config declaration, does not ensure the repository has been removed from the cluster, this is only to create a snapshot repo config. This is done to ensure that accidental removing and cleaning of snapshots is not done if re-applied with a missing entry. @segalziv @swoehrl-mw.